test/integration_test.py

```diff
@@ -724,8 +728,13 @@ def run(thread_count):
                 }
                 self.portal_service._crud(lambda db: [*db, mock_entry])
 
-        with ThreadPoolExecutor(max_workers=n_threads) as executor:
+        executor = ThreadPoolExecutor(max_workers=n_threads)
+        try:
             futures = [executor.submit(run, i) for i in range(n_tasks)]
+            executor.shutdown(wait=True)
```

Comment:
nit: wait=True is the default arg, could remove it.

Reply:
Point taken but personally I think the code benefits from being explicit about this. I could be wrong.